### ISSUE TITLE
Allow @property without dollar sign

### ISSUE THREAD

**Issue by daohoangson** [phpstan/phpdoc-parser#15]
Some of our libraries do not format their phpdoc property so phpstan is complaining about bad property access within our app code. Example: https://github.com/twilio/twilio-php/blob/master/Twilio/Rest/Client.php:

```php
/**
 * @property \Twilio\Rest\Api api <~ no dollar sign!
 */
```

Is it reasonable to make this parser support this kind of `@property` declaration? I think a simple fix to this line should be enough:

https://github.com/phpstan/phpdoc-parser/blob/ed3223362174b8067729930439e139794e9e514a/src/Lexer/Lexer.php#L146

**Comment by ondrejmirtes** [comment]
Nope, such format is invalid and should be fixed in the source. See https://docs.phpdoc.org/references/phpdoc/tags/property.html

**Comment by daohoangson** [comment]
Thanks @ondrejmirtes for getting back so fast. I understand that it's invalid. We have used class reflection extension to "patch" the library successfully but it's hurting adoption of phpstan within our org. Even the big names are doing it wrong, maybe we should relax a bit for real world usage?

**Comment by JanTvrdik** [comment]
@daohoangson It can be fixed with a single regexp replace across the entire code-base (can be done for example in PhpStorm, or pretty much any other editor).

**Comment by daohoangson** [comment]
I appreciate the responses, thank you both @JanTvrdik @ondrejmirtes.

**Comment by voku** [comment]
if someone found this via google and need a simple regex :)

---

find param without "$"

@param[\s]*([^$]*)$

---

find property without "$"

@property[\s]*([^$]*)$
